src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.pileupCallerDRAGEN.gatk4.vcf

```diff
@@ -145,11 +145,8 @@
 20	10098110	.	G	C	573.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=2.510;DP=36;ExcessHet=0.0000;FS=6.658;MLEAC=1;MLEAF=0.500;MQ=58.71;MQRankSum=-1.477;QD=15.93;ReadPosRankSum=-0.730;SOR=0.069	GT:AD:DP:GQ:PL	0/1:19,17:36:99:581,0,631
 20	10098135	.	C	A	436.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=-1.155;DP=39;ExcessHet=0.0000;FS=4.337;MLEAC=1;MLEAF=0.500;MQ=59.65;MQRankSum=1.546;QD=11.49;ReadPosRankSum=-0.591;SOR=0.075	GT:AD:DP:GQ:PL	0/1:24,14:38:99:444,0,874
 20	10098786	.	C	T	32.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=0.431;DP=9;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=54.65;MQRankSum=-2.200;QD=3.63;ReadPosRankSum=-1.383;SOR=0.132	GT:AD:DP:GQ:PL	0/1:7,2:9:40:40,0,217
-20	10099044	.	A	C	49.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=-1.201;DP=22;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=56.15;MQRankSum=-3.419;QD=2.76;ReadPosRankSum=-1.481;SOR=0.193	GT:AD:DP:GQ:PL	0/1:15,3:18:57:57,0,499
-20	10099046	.	T	C	73.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=-1.822;DP=18;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=55.26;MQRankSum=-3.419;QD=4.09;ReadPosRankSum=-1.364;SOR=0.193	GT:AD:DP:GQ:PL	0/1:15,3:18:81:81,0,561
-20	10099055	.	T	C	47.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=-1.460;DP=19;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=55.52;MQRankSum=-3.419;QD=2.65;ReadPosRankSum=-0.830;SOR=0.330	GT:AD:DP:GQ:PL	0/1:15,3:18:55:55,0,557
-20	10099079	.	C	T	40.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=0.694;DP=23;ExcessHet=0.0000;FS=2.192;MLEAC=1;MLEAF=0.500;MQ=56.33;MQRankSum=-3.254;QD=1.77;ReadPosRankSum=-0.568;SOR=1.460	GT:AD:DP:GQ:PL	0/1:19,4:23:48:48,0,551
-20	10099140	.	G	T	583.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=-0.176;DP=49;ExcessHet=0.0000;FS=6.794;MLEAC=1;MLEAF=0.500;MQ=60.07;MQRankSum=1.344;QD=12.42;ReadPosRankSum=-0.466;SOR=1.944	GT:AD:DP:GQ:PL	0/1:28,19:47:99:591,0,932
+20	10099046	.	T	C	31.64	.	AC=1;AF=0.500;AN=2;BaseQRankSum=-1.842;DP=17;ExcessHet=0.0000;FS=3.701;MLEAC=1;MLEAF=0.500;MQ=56.07;MQRankSum=-3.254;QD=1.86;ReadPosRankSum=-0.597;SOR=0.090	GT:AD:DP:GQ:PL	0/1:15,2:17:39:39,0,564
```

Comment:
these diferences are mostly noise to do with exactly what haplotypes come out... it is interesting the ties at the 5th score level seem to be very common... We really need to replace this heuristic with something not quite so terrible... @davidbenjamin

Reply:
@jamesemery It seems to be that the intent here is to corral the combinatorial explosion of paths in the de Bruijn graph that don't correspond to actual phasing.  However, by using the same kmer size as in assembly the heuristic is completely toothless!  It's basically a tautology that all the kmers of an assembled haplotype show up in some read, since otherwise the kmer wouldn't exist in the graph.

A better approach would be to use the same heuristic but with larger kmers than used in assembly.  For example, using k = 30 would get rid of spurious phasing within 30 bases.

Also, I don't think that using a larger k is merely a tie-breaker for the extant approach.  Rather, what we have now  should be replaced completely.

I don't think I am done pondering this but that's a good start.

Reply:
One might also ask whether an excess of haplotypes means that instead of filtering we should just redo assembly with a larger k, or dynamically switch to linked de Bruijn graphs.  That could well be.

Reply:
That seems like a good observation... It would also make it harder for every haplotype to be maximally represented by reads! I think it might also be worth considering normalizing to the lenght of the haplotype as well to prevent ourselves from being biased towards longer haplotypes that have incidental (and distant) insertions being over-represented in the choices here... alternatively we could flip the score on its head... where the score is the absolute number of missing kmers from the reads, which should more or less be capped at ~(2k -1) but not have this gross property that it prefers longer hapltotypes

Reply:
I agree that counting non-missing kmers and thereby rewarding length is wrong.

Since pileup haplotypes are constructed by injecting pileup events into base haplotypes, why not compute the score at that point as follows: 1) make a single kmer containing the pileup event and any nearby events on the base haplotype 2) check whether this kmer is supported in the reads.  If not, it's a spurious phasing. 3) If too many pileup haplotypes pass, repeat with larger k.

Reply:
@davidbenjamin i'm going to open a ticket to re-visit this heuristic... i think the band-aid here is sufficient and necessary to get into the next release.

Reply:
Agreed.